tracegen/internal/tracegen/config_test.go

```diff
@@ -0,0 +1,63 @@
+// Copyright The OpenTelemetry Authors
+// Copyright (c) 2018 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tracegen
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestKeyValueSet(t *testing.T) {
+	tests := []struct {
+		flag     string
+		expected KeyValue
+		err      error
+	}{
+		{
+			flag:     "key=\"value\"",
+			expected: KeyValue(map[string]string{"key": "value"}),
+		},
+		{
+			flag:     "key=\"\"",
+			expected: KeyValue(map[string]string{"key": ""}),
+		},
+		{
+			flag: "key=\"",
+			err:  errDoubleQuotesOTLPAttributes,
+		},
+		{
+			flag: "key=value",
```

Comment:
Aren't all the relevant parts parsing booleans from strings when needed anyway? I mean, error=true is semantically equivalent to error="true", and the values are likely to be converted to string anyway when going over the wire, especially when using a non-OTLP encoding. I'm not against this change, but I'm not quite sure the decrease in the user experience quality warrants it, as it's used only for two very specific values ("true" and "false").

Reply:
> Aren't all the relevant parts parsing booleans from strings when needed anyway? 

No, attributes passed through this option are [added as strings](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/b55861ba20a0ac48689c43a44ee90491a7d7a849/tracegen/main.go#L111).

> the values are likely to be converted to string anyway when going over the wire, especially when using a non-OTLP encoding

I don't understand this part: why does what happens over the wire matter?

> it's used only for two very specific values ("true" and "false").

This was a specific simple example, but the same applies to integers and floating point numbers (see [here](https://pkg.go.dev/go.opentelemetry.io/otel/attribute#KeyValue)), and in the Collector implementation we also support nested maps or the `null` value (see [here](https://pkg.go.dev/go.opentelemetry.io/collector/pdata@v0.54.0/internal#Map.InsertNull)), which even if not allowed by the current implementation, we may want to support in the future. 

I agree that it is a slight decrease in user quality, but I think the alternatives are worse: it's bad to set `key=true` or `key=400` and have to spend a lot of time to understand that these values were added as literal strings instead of the type that you would expect, and it's bad to break users in the future when adding support. Once we can parse all supported types, we could say "if this doesn't match any type, assume string", but for now we don't have support for any other type.

Reply:
> I don't understand this part: why does what happens over the wire matter?

Sorry, I mean that any server that is receiving this information should already be attempting to parse it in the formats that would make sense. The trace gen is a client, which could have been instrumented with another instrumentation library, encoding in another wire format and converting everything to string. In that case, the backend (perhaps not even the collector) would have to parse this value anyway before storing it.

In this particular case, I believe Jaeger would just store everything as string in the database and will parse the values for the special tag "error" when displaying it on the screen, adding the red marker if the value is "true" or true.

Reply:
In any case, I don't think we need to overcomplicate this, I'm fine in requiring quotes for strings.

Reply:
@jpkrohling anything you want me to change on this PR?